From 97439a7bc984db5d834f08f662d32bf48085a16c Mon Sep 17 00:00:00 2001
From: YangYonggang <b31664@freescale.com>
Date: Tue, 7 Feb 2012 17:41:15 +0800
Subject: [PATCH 233/502] ENGR00173958 Use GPU to do the preview surface
 composing instead of IPU.

Used GPU to do the preview surface composing instead of IPU.

Signed-off-by: YangYonggang <b31664@freescale.com>
---
 mx6/libcamera/CameraHal.cpp | 11 ++---------
 1 file changed, 2 insertions(+), 9 deletions(-)

diff --git a/mx6/libcamera/CameraHal.cpp b/mx6/libcamera/CameraHal.cpp
index 3ecd294..2ca95a3 100755
--- a/mx6/libcamera/CameraHal.cpp
+++ b/mx6/libcamera/CameraHal.cpp
@@ -757,14 +757,6 @@ namespace android {
             return err;
         }
 
-        err = mNativeWindow->set_usage(mNativeWindow, GRALLOC_USAGE_SW_READ_OFTEN |
-                GRALLOC_USAGE_SW_WRITE_OFTEN | GRALLOC_USAGE_HW_TEXTURE | GRALLOC_USAGE_HWC_OVERLAY);
-        if(err != 0){
-            CAMERA_HAL_ERR("native_window_set_usage failed:%s(%d)",
-                    strerror(-err), -err);
-            return err;
-        }
-
         int minUndequeueBufs = 0;
         err = mNativeWindow->get_min_undequeued_buffer_count(mNativeWindow,
                 &minUndequeueBufs);
@@ -812,7 +804,8 @@ namespace android {
 
             mCaptureBuffers[i].virt_start = (unsigned char *)handle->base;
             mCaptureBuffers[i].phy_offset = handle->phys;
-            mCaptureBuffers[i].length = handle->size;
+            //Calculate the buffer size, for GPU doesn't reply this value.
+            mCaptureBuffers[i].length =  mCaptureDeviceCfg.width * mCaptureDeviceCfg.height * 3 / 2;
             mCaptureBuffers[i].native_buf = (void *)buf;
             mCaptureBuffers[i].refCount = 0;
             mCaptureBuffers[i].buf_state = WINDOW_BUFS_DEQUEUED;
-- 
1.8.0

